src/main/java/seedu/foodrem/logic/parser/tagcommandparser/RenameTagCommandParser.java

```diff
@@ -41,6 +41,10 @@ public RenameTagCommand parse(String args) throws ParseException {
         final String originalName = matcher.group("originalName").trim();
         final String newName = matcher.group("newName").trim();
 
+        if (newName.equals(originalName)) {
+            throw new IllegalArgumentException(String.format("The tag name is already \"%s\".", newName));
```

Comment:
Can we use a `CommandException` here instead?

Reply:
After you mentioned this, I realised that this is command related logic and should be handled in the TagCommand Class instead of the parser. Have rectified it.